### ISSUE TITLE
v0.9.2 breaks prune operations

### ISSUE THREAD

**Issue by kapitainsky** [rustic-rs/rustic#1313]
When attempting to run prune:

```
$ rustic prune --instant-delete --max-unused 0% --max-repack=1GB
```

the following panic occurs:

```
$ cat /var/folders/36/02d6s6313zq5k3mfrhg4d8zc0000gn/T/report-f060cace-32eb-49a6-9b9f-fff65db481be.toml
name = "rustic-rs"
operating_system = "Mac OS 15.0.1 [64-bit]"
crate_version = "0.9.2"
explanation = """
Panic occurred in file '/Users/kptsky/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustic_core-0.5.2/src/blob/tree.rs' at line 698
"""
cause = """

                        Sending tree bc967136 on channel failed: sending on a full channel.

                        This should not happen! Please report it to the developers: https://github.com/rustic-rs/rustic_core/issues/new
                        """
method = "Panic"
backtrace = """

   0: 0x10a201e3a - __mh_execute_header
   1: 0x109d45aab - __mh_execute_header
   2: 0x109d437d4 - __mh_execute_header
   3: 0x10a0726ab - __mh_execute_header
   4: 0x7ff81a603253 - __pthread_start"""
```

It happens when rustic is searching for used blobs:

```
[00:00:00] finding used blobs...          ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░          0/36
```

reverting back to v0.9.1 solves the issue

**Comment by amorison** [comment]
The same error also occurs on `rustic check`, during the `checking trees...` step.

```toml
name = "rustic-rs"
operating_system = "Arch Linux [64-bit]"
crate_version = "0.9.2"
explanation = """
Panic occurred in file '/build/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustic_core-0.5.2/src/blob/tree.rs' at line 698
"""
cause = """

                        Sending tree ac08ce34 on channel failed: sending on a full channel. 
                        
                        This should not happen! Please report it to the developers: https://github.com/rustic-rs/rustic_core/issues/new
                        """
method = "Panic"
backtrace = """

   0: 0x590b781be684 - <unresolved>
   1: 0x590b784da61b - <unresolved>
   2: 0x7abefa9f639d - <unresolved>
   3: 0x7abefaa7b49c - <unresolved>
   4:        0x0 - <unresolved>"""
```

**Comment by simonsan** [comment]
Thanks for bringing it up! This is fixed in [rustic v0.9.3](https://github.com/rustic-rs/rustic/releases/tag/v0.9.3)!
